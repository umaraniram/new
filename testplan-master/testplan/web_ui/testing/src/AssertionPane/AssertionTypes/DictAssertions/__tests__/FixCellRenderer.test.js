import React from 'react';
import {shallow} from 'enzyme';
import {StyleSheetTestUtils} from "aphrodite";

import FixCellRenderer from '../FixCellRenderer';

function defaultProps() {
  return {};
}

function valueProps() {
  return {
    "value":{"value":"B","type":"B"},
    "data":{
      "descriptor":{
        "lineNo":667,
        "isFix":true,
        "indent":1,
        "isListKey":false,
        "isFailed":true
      },
      "key":{
        "value":601,
        "type":"key"
      },
      "value":{
        "value":"A",
        "type":"str"
      },
      "expected":{
        "value":"B",
        "type":"B"
      }
    },
    "colDef":{
      "headerName":"Expected",
      "field":"expected"
    }
  };
}


describe('DictLogAssertion', () => {
  let props;
  let shallowComponent;

  beforeEach(() => {
    // Stop Aphrodite from injecting styles, this crashes the tests.
    StyleSheetTestUtils.suppressStyleInjection();
    shallowComponent = undefined;
  });

  it('shallow renders the dict value', () => {
    props = valueProps();
    shallowComponent = shallow(<FixCellRenderer {...props}/>);
    expect(shallowComponent).toMatchSnapshot();
  });
  
});