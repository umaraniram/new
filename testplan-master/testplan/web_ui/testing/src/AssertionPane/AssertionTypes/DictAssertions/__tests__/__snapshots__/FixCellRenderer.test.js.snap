// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DictLogAssertion shallow renders the dict value 1`] = `
<Fragment>
  <DictCellRenderer
    data={
      Object {
        "descriptor": Object {
          "indent": 1,
          "isFailed": true,
          "isFix": true,
          "isListKey": false,
          "lineNo": 667,
        },
        "expected": Object {
          "type": "B",
          "value": "B",
        },
        "key": Object {
          "type": "key",
          "value": 601,
        },
        "value": Object {
          "type": "str",
          "value": "A",
        },
      }
    }
    id="id_667_undefined_undefined"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  />
</Fragment>
`;
